### PR TITLE
Add Candide Labs Social Recovery Module

### PR DESCRIPTION
**Pull request by nlordell**
This PR adds the Candide Labs social recovery module to the repository. It was done by importing the `candide-contracts` package at the specific audit commit, and then adding deployment scripts and documentations to support our release flows.

~~I'm waiting on the finalized audit report from Ackee to merge this.~~ **Edit**: final audit report has been added.

### CODE REVIEW

File: modules/recovery/package.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    "@types/yargs": "^17.0.32",
+    "dotenv": "^16.4.5",
+    "ethers": "^6.12.1",
+    "hardhat": "^2.22.3",
+    "hardhat-deploy": "^0.12.4",
+    "typescript": "^5.4.5",
+    "yargs": "^17.7.2"
+  },
+  "dependencies": {
+    "candide-contracts": "github:5afe/CandideWalletContracts#113d3c059e039e332637e8f686d9cbd505f1e738",
```
Comment (nlordell) [inline modules/recovery/package.json:52]: @andrewwahid - once we upstream the changes from our 5afe/CandideWalletContracts, we can depend on your contracts repository directly here as well.

File: modules/recovery/docs/v0.1.0/audit.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+## Notes
+
+The final audit was performed on commit [113d3c059e039e332637e8f686d9cbd505f1e738](https://github.com/5afe/CandideWalletContracts/tree/113d3c059e039e332637e8f686d9cbd505f1e738).
+
+All discovered findings were addressed.
+
+## Files
+
+- [Final audit report](audit-report-ackee-draft.pdf)
```
Comment (nlordell) [inline modules/recovery/docs/v0.1.0/audit.md:15]: Still waiting on audit report.

File: modules/recovery/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Changelog
+
+# Version 0.1.0
```
Comment (remedcu) [inline modules/recovery/CHANGELOG.md:3]: In the contract, they use `0.0.1`

https://github.com/5afe/CandideWalletContracts/blob/113d3c0/contracts/modules/social_recovery/SocialRecoveryModule.sol#L12

Comment (nlordell) [reply modules/recovery/CHANGELOG.md:3]: I would keep our versioning and their versioning separate.

File: modules/recovery/src/tasks/deploy_contracts.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { task } from 'hardhat/config'
+
+task('deploy-contracts', 'Deploys and verifies Safe contracts').setAction(async (_, hre) => {
+  await hre.run('deploy')
+  await hre.run('local-verify')
+  await hre.run('etherscan-verify', { forceLicense: true, license: 'LGPL-3.0' })
```
Comment (mmv08) [inline modules/recovery/src/tasks/deploy_contracts.ts:6]: Should id do sourcify too?

File: modules/recovery/src/utils/deployment.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export function getDeploymentParameters() {
```
Comment (mmv08) [inline modules/recovery/src/utils/deployment.ts:1]: nit: since this is very tightly related to the deployment I'd keep it in `src/deploy` folder

Comment (nlordell) [reply modules/recovery/src/utils/deployment.ts:1]: I just copied that pattern from the `safe-token`, but you are right that it isn't needed here (in the `safe-token`, these parameters were used for various tests as well, so it made sense to export them too).

Will move to the deployment script.

File: modules/recovery/src/tasks/deploy_contracts.ts (hunk 4 above)
Comment (nlordell) [reply modules/recovery/src/tasks/deploy_contracts.ts]: Added.

File: modules/recovery/src/utils/deployment.ts (hunk 5 above)
Comment (nlordell) [reply modules/recovery/src/utils/deployment.ts]: Changed.

File: .github/workflows/ci_recovery.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+      - uses: actions/checkout@v4
+      - uses: pnpm/action-setup@v3
+        with:
+          version: 9
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+          cache: pnpm
+      - run: |
+          pnpm install --frozen-lockfile
```
Comment (mmv08) [inline .github/workflows/ci_recovery.yml:20]: just fyi (i'm good with being explicit), looks like `--frozen-lockfile` is default in CI environment: https://pnpm.io/cli/install#--frozen-lockfile

**Review by mmv08 — APPROVED** [review]

File: .github/workflows/ci_recovery.yml (hunk 6 above)
Comment (nlordell) [reply .github/workflows/ci_recovery.yml:20]: Oh didn't know! I slightly prefer to be explicit (it also makes it a bit more reproducible when trying to debug "why did CI fail" on my local machine, and copy pasting commands from the workflow file).

**Pull request merged**